diff --git a/staging/src/k8s.io/apiserver/pkg/server/config.go b/staging/src/k8s.io/apiserver/pkg/server/config.go
index d10415b945f..112f065b4a6 100644
--- a/staging/src/k8s.io/apiserver/pkg/server/config.go
+++ b/staging/src/k8s.io/apiserver/pkg/server/config.go
@@ -421,6 +421,11 @@ func NewConfig(codecs serializer.CodecFactory) *Config {
 
 		hash := sha256.Sum256(hashData)
 		id = "apiserver-" + strings.ToLower(base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(hash[:16]))
+
+		prefix := os.Getenv("IDENTITY_PREFIX")
+		if prefix != "" {
+			id = prefix + "-" + id
+		}
 	}
 	lifecycleSignals := newLifecycleSignals()
 
