diff --git a/pkg/controlplane/apiserver/server.go b/pkg/controlplane/apiserver/server.go
index 26bf8c12..396e999a 100644
--- a/pkg/controlplane/apiserver/server.go
+++ b/pkg/controlplane/apiserver/server.go
@@ -55,17 +55,17 @@ import (
 var (
 	// IdentityLeaseGCPeriod is the interval which the lease GC controller checks for expired leases
 	// IdentityLeaseGCPeriod is exposed so integration tests can tune this value.
-	IdentityLeaseGCPeriod = 3600 * time.Second
+	IdentityLeaseGCPeriod = 5 * time.Second
 	// IdentityLeaseDurationSeconds is the duration of kube-apiserver lease in seconds
 	// IdentityLeaseDurationSeconds is exposed so integration tests can tune this value.
-	IdentityLeaseDurationSeconds = 3600
+	IdentityLeaseDurationSeconds = 30
 	// IdentityLeaseRenewIntervalPeriod is the interval of kube-apiserver renewing its lease in seconds
 	// IdentityLeaseRenewIntervalPeriod is exposed so integration tests can tune this value.
 	IdentityLeaseRenewIntervalPeriod = 10 * time.Second
 
 	// LeaseCandidateGCPeriod is the interval which the leasecandidate GC controller checks for expired leases
 	// This is exposed so integration tests can tune this value.
-	LeaseCandidateGCPeriod = 30 * time.Minute
+	LeaseCandidateGCPeriod = 1 * time.Minute
 )
 
 const (
