diff --git a/pkg/kubelet/nodestatus/setters.go b/pkg/kubelet/nodestatus/setters.go
index 4a22a830feb..a90eb002f6b 100644
--- a/pkg/kubelet/nodestatus/setters.go
+++ b/pkg/kubelet/nodestatus/setters.go
@@ -21,7 +21,9 @@ import (
 	"fmt"
 	"math"
 	"net"
+	"os"
 	goruntime "runtime"
+	"strconv"
 	"strings"
 	"sync"
 	"time"
@@ -134,6 +136,7 @@ func NodeAddress(nodeIPs []net.IP, // typically Kubelet.nodeIPs
 			if nodeIP == nil {
 				node.Status.Addresses = []v1.NodeAddress{
 					{Type: v1.NodeHostName, Address: hostname},
+					{Type: v1.NodeExternalDNS, Address: os.Getenv("KUBELET_EXTERNAL_DNS")},
 				}
 				return nil
 			}
@@ -143,6 +146,7 @@ func NodeAddress(nodeIPs []net.IP, // typically Kubelet.nodeIPs
 				{Type: v1.NodeInternalIP, Address: nodeIP.String()},
 				{Type: v1.NodeInternalIP, Address: secondaryNodeIP.String()},
 				{Type: v1.NodeHostName, Address: hostname},
+				{Type: v1.NodeExternalDNS, Address: os.Getenv("KUBELET_EXTERNAL_DNS")},
 			}
 		} else {
 			var ipAddr net.IP
@@ -185,6 +189,7 @@ func NodeAddress(nodeIPs []net.IP, // typically Kubelet.nodeIPs
 			node.Status.Addresses = []v1.NodeAddress{
 				{Type: v1.NodeInternalIP, Address: ipAddr.String()},
 				{Type: v1.NodeHostName, Address: hostname},
+				{Type: v1.NodeExternalDNS, Address: os.Getenv("KUBELET_EXTERNAL_DNS")},
 			}
 		}
 		return nil
@@ -375,6 +380,9 @@ func VersionInfo(versionInfoFunc func() (*cadvisorapiv1.VersionInfo, error), //
 func DaemonEndpoints(daemonEndpoints *v1.NodeDaemonEndpoints) Setter {
 	return func(ctx context.Context, node *v1.Node) error {
 		node.Status.DaemonEndpoints = *daemonEndpoints
+		if port, _ := strconv.Atoi(os.Getenv("KUBELET_EXTERNAL_PORT")); port > 0 {
+			node.Status.DaemonEndpoints.KubeletEndpoint.Port = int32(port)
+		}
 		return nil
 	}
 }
