diff --git a/pkg/features/kube_features.go b/pkg/features/kube_features.go
index e273afa59f3..0e9b9d08bdc 100644
--- a/pkg/features/kube_features.go
+++ b/pkg/features/kube_features.go
@@ -1957,7 +1957,7 @@ var defaultVersionedKubernetesFeatureGates = map[featuregate.Feature]featuregate
 
 	genericfeatures.APIServerIdentity: {
 		{Version: version.MustParse("1.20"), Default: false, PreRelease: featuregate.Alpha},
-		{Version: version.MustParse("1.26"), Default: true, PreRelease: featuregate.Beta},
+		{Version: version.MustParse("1.26"), Default: false, PreRelease: featuregate.Beta},
 	},
 
 	genericfeatures.APIServerTracing: {
diff --git a/staging/src/k8s.io/apiserver/pkg/features/kube_features.go b/staging/src/k8s.io/apiserver/pkg/features/kube_features.go
index 3c9a799a587..42f75c2319a 100644
--- a/staging/src/k8s.io/apiserver/pkg/features/kube_features.go
+++ b/staging/src/k8s.io/apiserver/pkg/features/kube_features.go
@@ -314,7 +314,7 @@ var defaultVersionedKubernetesFeatureGates = map[featuregate.Feature]featuregate
 
 	APIServerIdentity: {
 		{Version: version.MustParse("1.20"), Default: false, PreRelease: featuregate.Alpha},
-		{Version: version.MustParse("1.26"), Default: true, PreRelease: featuregate.Beta},
+		{Version: version.MustParse("1.26"), Default: false, PreRelease: featuregate.Beta},
 	},
 
 	APIServerTracing: {
