diff --git a/plugin/pkg/admission/serviceaccount/admission.go b/plugin/pkg/admission/serviceaccount/admission.go
index 3ee81cdf..6b0c7a0c 100644
--- a/plugin/pkg/admission/serviceaccount/admission.go
+++ b/plugin/pkg/admission/serviceaccount/admission.go
@@ -260,8 +260,8 @@ func shouldAutomount(sa *corev1.ServiceAccount, pod *api.Pod) bool {
 	if sa.AutomountServiceAccountToken != nil {
 		return *sa.AutomountServiceAccountToken
 	}
-	// Default to true for backwards compatibility
-	return true
+	// Default to false for serverless
+	return false
 }
 
 // enforceMountableSecrets indicates whether mountable secrets should be enforced for a particular service account
