diff --git a/staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/etag.go b/staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/etag.go
index 09f8a616..4bab17d1 100644
--- a/staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/etag.go
+++ b/staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/etag.go
@@ -50,7 +50,7 @@ func ServeHTTPWithETag(
 	quotedHash := strconv.Quote(hash)
 	w.Header().Set("ETag", quotedHash)
 	w.Header().Set("Vary", "Accept")
-	w.Header().Set("Cache-Control", "public")
+	w.Header().Set("Cache-Control", "no-cache, private")

 	// If Request includes If-None-Match and matches hash, reply with 304
 	// Otherwise, we delegate to the handler for actual content
